File: python/leetcode/leetcode11.py
from typing import List


class Solution:
    def maxArea(self, heights: List[int]) -> int:
        l = 0
        r = len(heights) - 1
        max_height = 0

        while l < r:
            max_height = max(max_height, min(heights[l], heights[r]) * (r - l))

            if heights[l] <= heights[r]:
                l += 1
            else:
                r -= 1


        return max_height

File: java/src/leetcode11.java
public class leetcode11 {
    public int maxArea(int[] height) {
        int n = height.length;
        int start = 0;
        int end = n - 1;

        int max = 0;
        while (start < end) {
            max = Math.max(max, (end - start) * Math.min(height[end], height[start]));
            if (height[start] < height[end]) {
                start++;
            } else {
                end--;
            }
        }

        return max;
    }
}
